testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java

```diff
@@ -3749,26 +3774,26 @@ void checkPeriodPredicate(Checker checker) {
         + "select * from e except "
         + "select * from f union "
         + "select * from g";
-    final String expected = "((((SELECT *\n"
+    final String expected = "SELECT *\n"
```

Comment:
Since we are changing `SET_QUERY` to non-expression, could you also add a test case to make sure the precedence for set operators could work well? such as `(select * from a union select * from b) intersect select * from c` should be unparsed as is.

Reply:
`SqlParserTest#testUnionIntersect` has been added. The reason I have not added CALCITE-5299's java doc on this test case is I think this test case is a complement for past precedence testing instead of for current CALCITE-5299.